### ISSUE TITLE
HH member array for 13+ members

### ISSUE THREAD

**Issue by CDPotter** [MN-Script-Team/BZS-FuncLib#26]
So it finally happened, a worker found a case that with 14 active HH members. As such the HH member array function left out memb 14. Is this something we should look at fixing?

**Comment by theVKC** [comment]
Maybe, but there’s a lot of other bugs and enhancements that should be worked on first. I don’t like seeing that list pile up…


Caution: This e-mail and attached documents, if any, may contain information that is protected by state or federal law. E-mail containing private or protected information should not be sent over a public (nonsecure) Internet unless it is encrypted pursuant to DHS standards. This e-mail should be forwarded only on a strictly need-to-know basis. If you are not the intended recipient, please: (1) notify the sender immediately, (2) do not forward the message, (3) do not print the message and (4) erase the message from your system.

**Comment by theVKC** [comment]
Hey @CDPotter, did you want to work on this?

**Comment by CDPotter** [comment]
Sure, I can work on it between things.

**Comment by CDPotter** [comment]
Hopefully there are others who see this. While trying to tackle this we think we found a clunky solution to grabbing all the membs. The issue arose with the lawlessness of the reference numbers and how they react to pf8ing. (see fake case 219585) When you pf8 it fills in the as many new reference numbers as it can so it you have 30 people the first screen is 1-13, pf8, 14-16, pf8, 18-30. Otherwise if you have 14 people it goes 1-13 pf8 2-14. 

I think the solution is to just transmit through memb creating an array and reading each panel (this will naturally slow down the CAF a bit.). Though using that method brought me to our next issue which is creating the dialog and variables so dynamically. 

Has anyone had to create unique variables on the fly?

For example if the script reads 30 HH members for the array the dialog needs to stretch 15 for each member and then create 30 unique variables and check boxes which can be used to determine the new array.

**Comment by theVKC** [comment]
You can do the 30 unique variables with an array like this (paraphrasing):
```
Dim array(0)
<Loop through to get the number of people>
ReDim array(number_of_people)
```

Then, array(0) will have the first person, array(1) the second, etc..

Frankly (my opinion), not sure if it's worth losing sleep over. It's a rare occurrence and I'd probably settle for a message telling the worker that it only grabbed info for membs 1-13 for the time being. But, it might be as simple as creating the array above and scrolling through MEMB only IF there's more HH members than 13...

**Comment by theVKC** [comment]
I've also found, when resizing dynamic dialogs, using UBound on that array helps. So something like `dialog_height = UBound(array) * 30`.

**Comment by theVKC** [comment]
Resolved with #66.
